sound/soc/sof/topology.c

```diff
@@ -2183,7 +2183,8 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 	process->comp.hdr.size = ipc_size;
 	process->comp.hdr.cmd = SOF_IPC_GLB_TPLG_MSG | SOF_IPC_TPLG_COMP_NEW;
 	process->comp.id = swidget->comp_id;
-	process->comp.type = type;
+	process->comp.type = SOF_COMP_PROCESS;
+	process->comp.subtype = type;
```

Comment:
@keyonjie Should this be conditional to FW version? If kernel is used with older firmware, this would break this case, right?

Reply:
Good point, let me add a check here.